### ISSUE TITLE
Navigo can't detect child path without declared parent path

### ISSUE THREAD

**Issue by ErcinDedeoglu** [krasimir/navigo#245]
Navigo can't detect path like "**/email-verification/x**" with this code:

```
        Mfy.Router.Navigo
            .on({
                'email-verification/:emailVerificationCode': function (emailVerificationCode) {
                    console.log("#Route: " + this.location.pathname);
                }
            })
            .resolve();
```

**But working when i put parent path:**

```
        Mfy.Router.Navigo
            .on({
                'email-verification': function () {
                    console.log("#Route: " + this.location.pathname);
                },
                'email-verification/:emailVerificationCode': function (emailVerificationCode) {
                    console.log("#Route: " + this.location.pathname);
                }
            })
            .resolve();
```

Is this a bug?

**Comment by ErcinDedeoglu** [comment]
https://github.com/krasimir/navigo/issues/177#issuecomment-351274375
this resolved issue but i bet this is a bug.
My old init:
`Navigo: new Navigo(null, false, null),`

**Comment by axebox** [comment]
To be honest, this is very basic feature that should be working by default. There really is no excuse for this to be broken here.

**Comment by krasimir** [comment]
Sorry for not responding here. The problem describe above is fixed in the latest version 8.0.0. Can you  try it out. It's covered with a unit test:

```js
describe("and the request described in #245", function () {
    it("should properly resolve the child path without the need of a parent path", function () {
      const router: NavigoRouter = new Navigo("/");
      const handler = jest.fn();

      router
        .on({
          "email-verification/:emailVerificationCode": handler,
        })
        .resolve("/email-verification/x");

      expect(handler).toBeCalledTimes(1);
      expect(handler).toBeCalledWith(
        expect.objectContaining({
          url: "email-verification/x",
          data: { emailVerificationCode: "x" },
        })
      );
    });
  });
```

Version 8.0.0 is not officially released but can be installed via `npm install navigo@beta` or `yarn add navigo@beta`. Migration guide is available here https://github.com/krasimir/navigo/blob/big-rewrite/CHANGELOG.md#migration-guide and the new documentation here https://github.com/krasimir/navigo/blob/big-rewrite/DOCUMENTATION.md.

P.S.S
I'm trying to start fresh on the new version 8.0.0. If you try it and still doesn't work let me know and I'll reopen the issue.
